AesCbcWithIntegrity.java

```diff
@@ -580,6 +581,21 @@ public boolean equals(Object obj) {
     }
 
     /**
+     * Copy the elements from the start to the end
+     *
+     * @param from  the source
+     * @param start the start index to copy
+     * @param end   the end index to finish
+     * @return the new buffer
+     */
+    private static byte[] copyOfRange(byte[] from, int start, int end) {
+        int length = end - start;
```

Comment:
I also think that this case would never happen with the current usage, since the method is not exposed to outside. Althought it never hurts to have validation check.
<code>copyOfRange(keyBytes, 0, 16)</code>
<code>copyOfRange(keyBytes, 16, 48;</code>